File: Front/src/models/BoardResponse.ts
import { UserStyleDto } from "./UserStyle";

export type ParentType = "post" | "comment";

export type User = {
    id: string;
    userName: string;
    style: UserStyleDto;
};

export type Reaction = {
    type: string;
    count: number;
    isMine: boolean;
};

export type Comment = {
    id: string;
    userId: string | null;
    text: string | null;
    createdAt: string;

    isEdited: boolean;
    isDeleted: boolean;
    isMine: boolean;

    reactions: Reaction[];
    replies: Comment[];
};

export type PostDto = {
    id: string;
    contentJson: string;
    category: string;
    createdAt: string;
    isPinned: boolean;
    reactions: Reaction[];
    comments: Comment[];
};

export type SectionType = "text" | "image";

export type PostSection = {
    type: SectionType;
    content: string;
    style?: Record<string, string>;
};

export type Post = {
    id: string;
    sections: PostSection[];
    category: string;
    createdAt: string;
    isPinned: boolean;
    reactions: Reaction[];
    comments: Comment[];
};

export type BoardDto = {
    posts: PostDto[];
    users: User[];
    totalPosts: number;
};

export type Board = {
    posts: Post[];
    users: User[];
    totalPosts: number;
};

File: Back/Responses/Board/BoardResponse.cs
public class BoardResponse
{
    public List<PostWithCommentsDto> Posts { get; set; } = [];
    public List<UserDto> Users { get; set; } = [];
    public int TotalPosts { get; set; }
}

public class PostWithCommentsDto
{
    public string Id { get; set; }
    public string ContentJson { get; set; }
    public string Category { get; set; }
    public DateTime CreatedAt { get; set; }
    public bool IsPinned { get; set; }

    public List<ReactionDto> Reactions { get; set; } = [];
    public List<CommentDto> Comments { get; set; } = [];
}

public class CommentDto
{
    public string Id { get; set; }
    public string? UserId { get; set; }
    public string? Text { get; set; }
    public DateTime CreatedAt { get; set; }

    public bool IsEdited { get; set; }
    public bool IsDeleted { get; set; }
    public bool IsMine { get; set; }

    public List<ReactionDto> Reactions { get; set; } = [];
    public List<CommentDto> Replies { get; set; } = [];
}

public class UserDto
{
    public string Id { get; set; }
    public string UserName { get; set; }
    public UserStyleDto Style { get; set; }
}

public class ReactionDto
{
    public string Type { get; set; }
    public int Count { get; set; }
    public bool IsMine { get; set; }
}
